src/data/notification.py

```diff
@@ -0,0 +1,8 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+
+class Notification(ABC):
+    @abstractmethod
+    def to_dict(self) -> Dict[str, Any]:  # pragma: no cover
```

Comment:
Could you cover this by having a test implemention of the class not implement this method?

Reply:
That's an interesting idea. My thought was that my test should verify a behaviour. Since the abstract method does not have any behaviour to verify I didn't see the value in a test for the abstract method. The implementation in a derived classes is tested here https://github.com/hmrc/platsec-compliance-alerting/pull/90/files#diff-08d511c9c58488429425eaf6c7da602338bc62722c76ec121e8874be3ef016cc